### DISCUSSION TITLE
Road to have `Multi-phased` extensions

### DISCUSSION THREAD

**Discussion by mkikets99** [krglaws/MyLFS#9] [Show and tell]
TODO (Will be updated):
 - [x] Move phases to separate folder.
 - [x] Move phases config to the `$PHASENAME/phases.config.sh` file
 - [x] Make config for extension by having `config.sh` file with information:
   - [x] How many phases does it have
   - [x] Make functions that runned before and after building 
- [x] Make a folder-like `static` and `templates` folders

If any question and suggestions - ask.

**Comment by krglaws** [comment]
Just a heads up: I just pushed a fix for an issue in the `check_dependencies` function, but I think it should be an easy merge since it doesn't seem like you will need to touch that part of the script.

**Reply by mkikets99** [reply]
I upstreamed it on fork... Automerge was successful.
I am continuing to work on it...

Also.. about idea with `systemd`... if this will work correctly, it can be used to build either with `sysvinit` or `systemd`. Just different `--phase-id` will be used...

... if you dont like `-t|--phase-id` spelling you can provide your option

**Reply by krglaws** [reply]
Want to create a PR? It's ok if it isn't ready yet, you can continue to work on it, but having a PR open would help us to discuss the changes, and I'd be able to look at the diff and see how it looks.

**Reply by mkikets99** [reply]
Done...
Also i think i may finished this portion of scripting...
Now i will check why it not executing grub and syslinux

**Reply by mkikets99** [reply]
Also: I think that `build_extension` is now unnecessary because of new `--phase-id` option...
